packages/core/src/internal/middleware.js

```diff
@@ -15,15 +14,12 @@ export default function sagaMiddlewareFactory({ context = {}, ...options } = {})
       check(onError, is.func, 'options.onError passed to the Saga middleware is not a function!')
     }
 
-    if (is.notUndef(options.emitter)) {
-      check(options.emitter, is.func, 'options.emitter passed to the Saga middleware is not a function!')
+    if (is.notUndef(channel)) {
```

Comment:
minor thing. should we use `is.channel` instead?

Reply:
I'm using the specific checks below if options.channel if defined.  I'v found a small bug in `is.channel` below.